foreign_cc/built_tools/make_build.bzl

```diff
@@ -29,8 +36,69 @@ def _make_tool_impl(ctx):
             "cp -p ./{}/gnumake.exe $$INSTALLDIR$$/bin/make.exe".format(dist_dir),
         ]
     else:
+        feature_configuration = cc_common.configure_features(
+            ctx = ctx,
+            cc_toolchain = cc_toolchain,
+            requested_features = ctx.features,
+            unsupported_features = ctx.disabled_features,
+        )
+        ar_path = cc_common.get_tool_for_action(
+            feature_configuration = feature_configuration,
+            action_name = CPP_LINK_STATIC_LIBRARY_ACTION_NAME,
+        )
+        cc_path = cc_common.get_tool_for_action(
+            feature_configuration = feature_configuration,
+            action_name = C_COMPILE_ACTION_NAME,
+        )
+        ld_path = cc_common.get_tool_for_action(
+            feature_configuration = feature_configuration,
+            action_name = CPP_LINK_EXECUTABLE_ACTION_NAME,
+        )
+        c_compile_variables = cc_common.create_compile_variables(
+            feature_configuration = feature_configuration,
+            cc_toolchain = cc_toolchain,
+        )
+        cflags = cc_common.get_memory_inefficient_command_line(
+            feature_configuration = feature_configuration,
+            action_name = C_COMPILE_ACTION_NAME,
+            variables = c_compile_variables,
+        )
+        c_link_variables = cc_common.create_link_variables(
+            feature_configuration = feature_configuration,
+            cc_toolchain = cc_toolchain,
+        )
+        ldflags = cc_common.get_memory_inefficient_command_line(
+            feature_configuration = feature_configuration,
+            action_name = CPP_LINK_EXECUTABLE_ACTION_NAME,
+            variables = c_link_variables,
+        )
+
+        sysroot_cflags = [flag for flag in cflags if flag.startswith("--sysroot=")]
+        non_sysroot_cflags = [flag for flag in cflags if not flag.startswith("--sysroot=")]
+        sysroot_ldflags = [flag for flag in ldflags if flag.startswith("--sysroot=")]
+        non_sysroot_ldflags = [flag for flag in ldflags if not flag.startswith("--sysroot=")]
+
+        # Make's build script does not forward CFLAGS to all compiler and linker
+        # invocations, so we append --sysroot flags directly to CC.
+        absolute_cc = _absolutize(ctx.workspace_name, cc_path, True)
+        if sysroot_cflags:
+            absolute_cc += " " + _join_flags_list(ctx.workspace_name, sysroot_cflags)
+        absolute_ld = _absolutize(ctx.workspace_name, cc_path, True)
+        if sysroot_ldflags:
+            absolute_ld += " " + _join_flags_list(ctx.workspace_name, sysroot_ldflags)
+
+        env = {
+            "AR": _absolutize(ctx.workspace_name, ar_path, True),
```

Comment:
To resolve the build failure on MacOS, you may have to unset the `AR` variable, as is done in a few of the third party examples in this repo

Reply:
To be honest I don’t really know why `AR` has to be unset for MacOS. Perhaps @UebelAndre or @jsharpe could offer some insight?

Reply:
I'm not entirely sure either but suspect its an incompatibility between setting ARFLAGS to be the output from the cc toolchain's `cxx_linker_static`  probably because the native link static rule adds additional flags somewhere else but I don't know the mechanics of the toolchain on macOS well enough to be able to say that with complete certainty.

@fmeum would you mind separating this PR into 2 to make the changes to `AR` on macOS separately so that it can be reverted independently if it causes any breakage that the examples in this repo don't show?

Reply:
Oh wait - you already did - my bad; I should read the full PR before commenting!

Reply:
Sort of - I still kept the change to `AR` in this PR for now, but would wait until I got #819 to work and merged. Afterwards, I would be able to delete the logic from this PR.